rotkehlchen/tests/api/test_history.py

```diff
@@ -128,7 +128,7 @@ def test_query_history(rotkehlchen_api_server_with_exchanges, start_ts, end_ts):
         ),
     )
     result = assert_proper_response_with_result(response=response, status_code=HTTPStatus.OK)
-    assert len(result['missing_acquisitions']) == 10
+    assert len(result['missing_acquisitions']) == 9
```

Comment:
why is there one less?

Reply:
The problem is that in the mocks that we have here there are many events that occur on the same timestamp (`1458994442`). Before this PR, events were added in batches per-exchange (e.g. queried events for exchange №1 -> added these events, then queried events for exchange №2 -> added these events).

With this pr's changes, now all events are added all at once and since we order only by timestamp, the order is a bit different.

Reply:
Mmmmmm but then I guess this is a bug we had before somehow? Is not exactly a bug but uncovers an issue on that logic :thinking:

Reply:
order being a bit different is one thing, but missing an event is another. How is an event missing? Something else is off here and should be looked at.

Or you mean that the order of events in processing changed so somehow there is one less missing acquisition. If that is so then I would need you to show me exactly which event lead to this otherwise I would be afraid there is something more sinister underneath.

I guess you already spent time doing that so can you point me to that event whose order change did it? Perhaps easiest way to do is run a certain test with debugger on for both this PR and bugixes/develop.

Reply:
[initial-events-old.txt](https://github.com/rotki/rotki/files/10277609/initial-events-old.txt)
[initial-events-new.txt](https://github.com/rotki/rotki/files/10277610/initial-events-new.txt)
[processed-events-old.txt](https://github.com/rotki/rotki/files/10277672/processed-events-old.txt)
[processed-events-new.txt](https://github.com/rotki/rotki/files/10277613/processed-events-new.txt)

I attached 4 json files (they have .txt extension here since github doesn't support json files).

`initial-` files - events before accounting
`processed-` files - processed events

`-old` files - before this pr changes
`-new` files - after this pr changes

In `initial-events-old` you can see these 2 events in one order:
```json
{
        "identifier": "f0bdeb714921a9d8dd14f6ce389c710ad4564fc8c0e94b2c12ba57f87719ad2b",
        "timestamp": 1458994442,
        "location": "poloniex",
        "category": "withdrawal",
        "address": "131rdg5Rzn6BFufnnQaHhVa5ZtRU1J2EZR",
        "transaction_id": "2d27ae26fa9c70d6709e27ac94d4ce2fde19b3986926e9f3bfcf3e2d68354ec5",
        "asset": "BTC",
        "amount": "5.0",
        "fee_asset": "BTC",
        "fee": "0.5",
        "link": "1"
    },
    {
        "timestamp": 1458994442,
        "location": "kraken",
        "base_asset": "BTC",
        "quote_asset": "EUR",
        "trade_type": "buy",
        "amount": "1",
        "rate": "100",
        "fee": "0.1",
        "fee_currency": "EUR",
        "link": "11458994442",
        "notes": null
    },
```

while in `initial-events-new` the order is reversed:
```json
{
        "timestamp": 1458994442,
        "location": "kraken",
        "base_asset": "BTC",
        "quote_asset": "EUR",
        "trade_type": "buy",
        "amount": "1",
        "rate": "100",
        "fee": "0.1",
        "fee_currency": "EUR",
        "link": "11458994442",
        "notes": null
    },
    {
        "identifier": "f0bdeb714921a9d8dd14f6ce389c710ad4564fc8c0e94b2c12ba57f87719ad2b",
        "timestamp": 1458994442,
        "location": "poloniex",
        "category": "withdrawal",
        "address": "131rdg5Rzn6BFufnnQaHhVa5ZtRU1J2EZR",
        "transaction_id": "2d27ae26fa9c70d6709e27ac94d4ce2fde19b3986926e9f3bfcf3e2d68354ec5",
        "asset": "BTC",
        "amount": "5.0",
        "fee_asset": "BTC",
        "fee": "0.5",
        "link": "1"
    },
```

As the result of this in `processed-events-old` poloniex withdrawal event has no matched acquisitions
```json
{
            "type": "asset movement",
            "notes": "poloniex withdrawal",
            "location": "poloniex",
            "timestamp": 1458994442,
            "asset": "BTC",
            "free_amount": "0.0",
            "taxable_amount": "0.5",
            "price": "373.295",
            "pnl_taxable": "0",
            "pnl_free": "0",
            "cost_basis": {
                "is_complete": false,
                "matched_acquisitions": []
            }
        },
```

while in `processed-events-new` it does:
```json
{
            "type": "asset movement",
            "notes": "poloniex withdrawal",
            "location": "poloniex",
            "timestamp": 1458994442,
            "asset": "BTC",
            "free_amount": "0.0",
            "taxable_amount": "0.5",
            "price": "373.295",
            "pnl_taxable": "0",
            "pnl_free": "0",
            "cost_basis": {
                "is_complete": true,
                "matched_acquisitions": [
                    {
                        "amount": "0.5",
                        "event": {
                            "timestamp": 1458994442,
                            "full_amount": "1",
                            "rate": "100.1",
                            "index": 10
                        },
                        "taxable": true
                    }
                ]
            }
```

Reply:
Okay that makes sense thank you.